metrics/all_kind.go

```diff
@@ -0,0 +1,99 @@
+	prommHandler    http.Handler
+	codaHaleHandler http.Handler
+}
+
+func NewAll(o Options) *All {
+	return &All{
+		prometheus: NewPrometheus(o),
+		codaHale:   NewCodaHale(o),
+	}
+}
+
+func (a *All) MeasureSince(key string, start time.Time) {
+	a.prometheus.MeasureSince(key, start)
+	a.codaHale.MeasureSince(key, start)
+}
+func (a *All) IncCounter(key string) {
+	a.prometheus.IncCounter(key)
+	a.codaHale.IncCounter(key)
+
+}
+func (a *All) MeasureRouteLookup(start time.Time) {
+	a.prometheus.MeasureRouteLookup(start)
+	a.codaHale.MeasureRouteLookup(start)
+}
+func (a *All) MeasureFilterRequest(filterName string, start time.Time) {
+	a.prometheus.MeasureFilterRequest(filterName, start)
+	a.codaHale.MeasureFilterRequest(filterName, start)
+}
+func (a *All) MeasureAllFiltersRequest(routeId string, start time.Time) {
+	a.prometheus.MeasureAllFiltersRequest(routeId, start)
+	a.codaHale.MeasureAllFiltersRequest(routeId, start)
+}
+func (a *All) MeasureBackend(routeId string, start time.Time) {
+	a.prometheus.MeasureBackend(routeId, start)
+	a.codaHale.MeasureBackend(routeId, start)
+}
+func (a *All) MeasureBackendHost(routeBackendHost string, start time.Time) {
+	a.prometheus.MeasureBackendHost(routeBackendHost, start)
+	a.codaHale.MeasureBackendHost(routeBackendHost, start)
+}
+func (a *All) MeasureFilterResponse(filterName string, start time.Time) {
+	a.prometheus.MeasureFilterResponse(filterName, start)
+	a.codaHale.MeasureFilterResponse(filterName, start)
+}
+func (a *All) MeasureAllFiltersResponse(routeId string, start time.Time) {
+	a.prometheus.MeasureAllFiltersResponse(routeId, start)
+	a.codaHale.MeasureAllFiltersResponse(routeId, start)
+}
+func (a *All) MeasureResponse(code int, method string, routeId string, start time.Time) {
+	a.prometheus.MeasureResponse(code, method, routeId, start)
+	a.codaHale.MeasureResponse(code, method, routeId, start)
+}
+func (a *All) MeasureServe(routeId, host, method string, code int, start time.Time) {
+	a.prometheus.MeasureServe(routeId, host, method, code, start)
+	a.codaHale.MeasureServe(routeId, host, method, code, start)
+}
+func (a *All) IncRoutingFailures() {
+	a.prometheus.IncRoutingFailures()
+	a.codaHale.IncRoutingFailures()
+}
+func (a *All) IncErrorsBackend(routeId string) {
+	a.prometheus.IncErrorsBackend(routeId)
+	a.codaHale.IncErrorsBackend(routeId)
+}
+func (a *All) MeasureBackend5xx(t time.Time) {
+	a.prometheus.MeasureBackend5xx(t)
+	a.codaHale.MeasureBackend5xx(t)
+
+}
+func (a *All) IncErrorsStreaming(routeId string) {
+	a.prometheus.IncErrorsStreaming(routeId)
+	a.codaHale.IncErrorsStreaming(routeId)
+
+}
+func (a *All) RegisterHandler(path string, handler *http.ServeMux) {
+	a.prommHandler = a.prometheus.CreateHandler()
+	a.codaHaleHandler = a.codaHale.CreateHandler(path)
+	handler.Handle(path, a.newHandler())
+}
+
+func (a *All) newHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Header.Get("Accept") == "application/json" {
```

Comment:
This seems like a hidden feature to me that can easily get lost. Any specific reason why it is done like that?

Reply:
Do you have another suggestion how to do this?
For me it was the native http way of doing it.

Reply:
It is and you are not totally wrong, but it seems a bit hidden to me... in the end `application/json` doesn't map necessarily to `codehale`, but just to the json format, you know what I mean?

Reply:
I understand, but what else do you recommend?

Reply:
Thinking about it twice, it is not that bad and I think actually the problem is just docs. I would do the following: 

- document the way metrics are exposed in Skipper, by saying that we use two formats.
- have flags such that it is always clear which metrics are enabled by default and how to use them. 

Also: is it actually right that in case you have `application/json` you do `codahale` and in the else prometheus? If I write `application/foobar` I get prometheus text format, is that intended? If so, it kinda clashes with the idea that `codahale` is the default... when the flags are not set for example you always get `codahale`, when they are both set, you have to be "very specific" to get `codahale` otherwise you get `prometheus`.

Reply:
i am thinking about using different paths for the different metrics. E.g:

/ -> the current default
/codahale -> codahale if enabled or 404
/prometheus -> prometheus if enabled or 404

not sure if the above is the best, but i agree with @Raffo 's comment that the mime type doesn't represent the types that we're choosing from.

Reply:
I had the same idea as @aryszka but then I didn't write it, I felt like I didn't want to be that specific to even put it in the path.

Reply:
If you want to go the content-type route I'd suggest doing something explicit like `application/codahale+json`.

Reply:
@aryszka the path change will also break the current monitoring, but content-type already does the same so for me it's not a problem to do the path routing approach.
Anyone against the path routing?
Someone prefer the Content-Type based solution?

Reply:
+1 for @aermakov-zalando

Reply:
@szuecs i also like @aermakov-zalando 's suggestion best now.

Reply:
I also talked with @mikkeloscar and we agreed to use application/codahale+json Content-Type based solution